blockdag/ghostdag.go

```diff
@@ -0,0 +1,166 @@
+package blockdag
+
+import (
+	"github.com/pkg/errors"
+	"sort"
+)
+
+func (dag *BlockDAG) selectedParentAnticone(node *blockNode) ([]*blockNode, error) {
+	anticoneSet := newSet()
+	var anticoneSlice []*blockNode
+	selectedParentPast := newSet()
+	var queue []*blockNode
+	for _, parent := range node.parents {
+		if parent == node.selectedParent {
+			continue
+		}
+		anticoneSet.add(parent)
+		anticoneSlice = append(anticoneSlice, parent)
+		queue = append(queue, parent)
+	}
+	for len(queue) > 0 {
+		var current *blockNode
```

Comment:
Maybe extract the handling of each parent to another function to improve readability.